routehandlers/handler.go

```diff
@@ -263,7 +263,12 @@ func (handler *Handler) handleActualUpdate(logger lager.Logger, before, after *m
 	)
 	switch {
 	case after.State == models.ActualLRPStateRunning:
-		routeMappings, messagesToEmit = handler.routingTable.AddEndpoint(logger, after)
+		if after.Routable == true {
+			routeMappings, messagesToEmit = handler.routingTable.AddEndpoint(logger, after)
+		} else if before.Routable == true {
```

Comment:
For other unregistration messages the [route emitter sends them multiple times](https://github.com/cloudfoundry/route-emitter/blob/a17e814f0e2741d298c84de3fe9821475dd67cdb/unregistration/sender_test.go#L78-L90).

Do these unregistration messages hook into that same logic and send multiple times (but only if the app hasn't been marked routable again!)? If we don't, then there is a high likelihood of some gorouters missing the unregistration message.

Reply:
The route emitter is still incorrectly only sending unregistration messages once. It should send them 10 times or until the app is ready again.

**Reproduction Steps**
* add add this hc to the proxy app
```
[
  {
    "http_check": {
      "path": "/eventuallyfail",
      "port": 8080
    }
  }
]
```
* get on a bosh VM and subscribe to nats 
```
wget https://github.com/nats-io/natscli/releases/download/v0.0.32/nats-0.0.32-linux-amd64.zip
unzip nats-0.0.32-linux-amd64.zip
chmod +x nats-0.0.32-linux-amd64
mv nats-0.0.32-linux-amd64/nats /usr/bin

nats sub "router.unregister" \
-s nats://NATS_USERNAME:NATS_PASSWORD@NATS_ADDRESS \
--tlsca NATS_CA_CERT_FILE \
--tlscert NATS_CLIENT_CERT_FILE \
--tlskey NATS_CLIENT_KEY_FILE
```
* see that there is only one unregistration message for the proxy app

Reply:
@ameowlia I was trying to understand how the logic is different from regular app events. I just pushed a simple app and after running `cf stop appa` I am only seeing one `router.unregister` event.

Reply:
If you **stop** an app it only sends the unregistration message once. If a gorouter misses this message and sends traffic to a stopped app it will fail and will prune the endpoint and will retry if there are other available backends.

If you **unmap** a route from an app, it sends the unregistration message multiple (10??) times. If a gorouter misses this message and sends traffic to this route, then the end user will be able to incorrectly access the app.

I think this case with the readiness healthcheck should work like unmapping a route rather than stopping an app.